[Automation] native-image run fails for org.springframework.boot:spring-boot-kafka:4.1.0

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27316664203
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
      "parameterTypes": [
           "org.springframework.kafka.core.ProducerFactory",
           "org.springframework.kafka.support.ProducerListener",
           "org.springframework.beans.factory.ObjectProvider",
           "org.springframework.beans.factory.ObjectProvider"
         ]
       }
     ]
   }
 
 The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
 	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
 	at java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:149)
 	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:152)
 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
 	... 61 more

       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$autoConfigurationBindsPropertiesAndCreatesKafkaInfrastructureBeans$0(Spring_boot_kafkaTest.java:135)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.accept(AbstractApplicationContextRunner.java:457)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.consumeAssertableContext(AbstractApplicationContextRunner.java:371)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$1(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.util.TestPropertyValues.lambda$applyToSystemProperties$0(TestPropertyValues.java:176)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:190)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:175)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$0(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.withContextClassLoader(AbstractApplicationContextRunner.java:377)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.run(AbstractApplicationContextRunner.java:348)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:factoryCustomizersContributeAdditionalClientConfiguration()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'factoryCustomizersContributeAdditionalClientConfiguration', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$factoryCustomizersContributeAdditionalClientConfiguration$6(Spring_boot_kafkaTest.java:212)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
    "methods": [
      {
        "name": "kafkaTemplate",
        "parameterTypes": [
          "org.springframework.kafka.core.ProducerFactory",
          "org.springframework.kafka.support.ProducerListener",
          "org.springframework.beans.factory.ObjectProvider",
          "org.springframework.beans.factory.ObjectProvider"
        ]
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
    "methods": [
      {
        "name": "kafkaTemplate",
        "parameterTypes": [
          "org.springframework.kafka.core.ProducerFactory",
          "org.springframework.kafka.support.ProducerListener",
          "org.springframework.beans.factory.ObjectProvider",
          "org.springframework.beans.factory.ObjectProvider"
        ]
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:149)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories$7(Spring_boot_kafkaTest.java:231)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
    "methods": [
      {
        "name": "kafkaTemplate",
        "parameterTypes": [
          "org.springframework.kafka.core.ProducerFactory",
          "org.springframework.kafka.support.ProducerListener",
          "org.springframework.beans.factory.ObjectProvider",
          "org.springframework.beans.factory.ObjectProvider"
        ]
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
    "methods": [
      {
        "name": "kafkaTemplate",
        "parameterTypes": [
          "org.springframework.kafka.core.ProducerFactory",
          "org.springframework.kafka.support.ProducerListener",
          "org.springframework.beans.factory.ObjectProvider",
          "org.springframework.beans.factory.ObjectProvider"
        ]
      }
    ]
  }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:149)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]

Test run finished after 34 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-kafka/4.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-kafka:4.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 36s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.1.0 at 2026-06-11T01:26:08Z with exit code 1 after 157s
❌ ERROR: 'native-image run' for 4.1.0 failed with exit code 1.
FAILED[native-image run][4.1.0][env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27316664203
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
g.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
 	at java.base@25.0.3/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
 	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:207)
 	... 46 more
 Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:
 
     {
       "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
       "methods": [
         {
           "name": "kafkaTemplate",
           "parameterTypes": [
             "org.springframework.kafka.core.ProducerFactory",
             "org.springframework.kafka.support.ProducerListener",
             "org.springframework.beans.factory.ObjectProvider",
             "org.springframework.beans.factory.ObjectProvider"
           ]
         }
       ]
     }
 
 The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
 	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
 	at java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
 	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:152)
 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
 	... 63 more

       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$autoConfigurationBindsPropertiesAndCreatesKafkaInfrastructureBeans$0(Spring_boot_kafkaTest.java:135)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.accept(AbstractApplicationContextRunner.java:457)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.consumeAssertableContext(AbstractApplicationContextRunner.java:371)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$1(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.util.TestPropertyValues.lambda$applyToSystemProperties$0(TestPropertyValues.java:176)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:190)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:175)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$0(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.withContextClassLoader(AbstractApplicationContextRunner.java:377)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.run(AbstractApplicationContextRunner.java:348)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:factoryCustomizersContributeAdditionalClientConfiguration()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'factoryCustomizersContributeAdditionalClientConfiguration', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$factoryCustomizersContributeAdditionalClientConfiguration$6(Spring_boot_kafkaTest.java:212)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories$7(Spring_boot_kafkaTest.java:231)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]

Test run finished after 35 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-kafka/4.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-kafka:4.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 37s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.1.0 at 2026-06-11T01:26:18Z with exit code 1 after 158s
❌ ERROR: 'native-image run' for 4.1.0 failed with exit code 1.
FAILED[native-image run][4.1.0][env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27316664203
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
tils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
 	at java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
 	at java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:152)
 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653)
 	... 63 more

       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$autoConfigurationBindsPropertiesAndCreatesKafkaInfrastructureBeans$0(Spring_boot_kafkaTest.java:135)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.accept(AbstractApplicationContextRunner.java:457)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.consumeAssertableContext(AbstractApplicationContextRunner.java:371)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$1(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.util.TestPropertyValues.lambda$applyToSystemProperties$0(TestPropertyValues.java:176)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:190)
       org.springframework.boot.test.util.TestPropertyValues.applyToSystemProperties(TestPropertyValues.java:175)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.lambda$run$0(AbstractApplicationContextRunner.java:349)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.withContextClassLoader(AbstractApplicationContextRunner.java:377)
       org.springframework.boot.test.context.runner.AbstractApplicationContextRunner.run(AbstractApplicationContextRunner.java:348)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:factoryCustomizersContributeAdditionalClientConfiguration()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'factoryCustomizersContributeAdditionalClientConfiguration', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$factoryCustomizersContributeAdditionalClientConfiguration$6(Spring_boot_kafkaTest.java:212)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]
  JUnit Jupiter:Spring_boot_kafkaTest:customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories()
    MethodSource [className = 'org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest', methodName = 'customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories', methodParameterTypes = '']
    => java.lang.IllegalStateException: Unstarted application context org.springframework.boot.test.context.assertj.AssertableApplicationContext[startupFailure=org.springframework.beans.factory.BeanCreationException] failed to start
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.getStartedApplicationContext(AssertProviderApplicationContextInvocationHandler.java:159)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invokeApplicationContextMethod(AssertProviderApplicationContextInvocationHandler.java:150)
       org.springframework.boot.test.context.assertj.AssertProviderApplicationContextInvocationHandler.invoke(AssertProviderApplicationContextInvocationHandler.java:87)
       jdk.proxy4/jdk.proxy4.$Proxy/s71c21bd4.getBean(Unknown Source)
       org_springframework_boot.spring_boot_kafka.Spring_boot_kafkaTest.lambda$customConnectionDetailsOverridePropertyBasedBootstrapServersForClientFactories$7(Spring_boot_kafkaTest.java:231)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration': Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321)
       org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1380)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1219)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration]: Constructor threw exception
       org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:220)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:129)
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318)
       [...]
     Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaTemplate' defined in org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:657)
       org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:645)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1360)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1192)
       org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:565)
       [...]
     Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: Cannot reflectively invoke method 'org.springframework.kafka.core.KafkaTemplate org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration.kafkaTemplate(org.springframework.kafka.core.ProducerFactory,org.springframework.kafka.support.ProducerListener,org.springframework.beans.factory.ObjectProvider,org.springframework.beans.factory.ObjectProvider)'. To allow this operation, add the following to the 'reflection' section of 'reachability-metadata.json' and rebuild the native image:

    {
      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
      "methods": [
        {
          "name": "kafkaTemplate",
          "parameterTypes": [
            "org.springframework.kafka.core.ProducerFactory",
            "org.springframework.kafka.support.ProducerListener",
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-kafka/4.0.2/build/native/nativeTestCompile/org.springframework.boot.spring-boot-kafka_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 33s

            "org.springframework.beans.factory.ObjectProvider",
            "org.springframework.beans.factory.ObjectProvider"
          ]
        }
      ]
    }

The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection
       org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.reportInvokedExecutable(MissingReflectionRegistrationUtils.java:110)
       java.base@25.0.3/java.lang.reflect.Method.acquireMethodAccessor(Method.java:151)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:562)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.lambda$instantiate$0(SimpleInstantiationStrategy.java:155)
       org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiateWithFactoryMethod(SimpleInstantiationStrategy.java:72)
       [...]

Test run finished after 47 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         3 tests successful      ]
[         4 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.springframework.boot/spring-boot-kafka/4.0.2/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.springframework.boot:spring-boot-kafka:4.0.2 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2m 35s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 4.1.0 at 2026-06-11T01:26:14Z with exit code 1 after 155s
❌ ERROR: 'native-image run' for 4.1.0 failed with exit code 1.
FAILED[native-image run][4.1.0][env GVM_TCK_LV=4.1.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.springframework.boot:spring-boot-kafka:4.0.2]
```

--- a/metadata/org.springframework.boot/spring-boot-kafka/4.1.0/reachability-metadata.json
+++ b/metadata/org.springframework.boot/spring-boot-kafka/4.1.0/reachability-metadata.json
@@ -1,0 +1,395 @@
+{
+  "reflection": [
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAutoConfiguration",
+      "allDeclaredConstructors": true,
+      "methods": [
+        {
+          "name": "kafkaConnectionDetails",
+          "parameterTypes": [
+            "org.springframework.beans.factory.ObjectProvider"
+          ]
+        },
+        {
+          "name": "kafkaTemplate",
+          "parameterTypes": [
+            "org.springframework.kafka.core.ProducerFactory",
+            "org.springframework.kafka.support.ProducerListener",
+            "org.springframework.beans.factory.ObjectProvider",
+            "org.springframework.beans.factory.ObjectProvider"
+          ]
+        },
+        {
+          "name": "kafkaProducerListener",
+          "parameterTypes": []
+        },
+        {
+          "name": "kafkaConsumerFactory",
+          "parameterTypes": [
+            "org.springframework.boot.kafka.autoconfigure.KafkaConnectionDetails",
+            "org.springframework.beans.factory.ObjectProvider"
+          ]
+        },
+        {
+          "name": "kafkaProducerFactory",
+          "parameterTypes": [
+            "org.springframework.boot.kafka.autoconfigure.KafkaConnectionDetails",
+            "org.springframework.beans.factory.ObjectProvider"
+          ]
+        },
+        {
+          "name": "kafkaTransactionManager",
+          "parameterTypes": [
+            "org.springframework.kafka.core.ProducerFactory"
+          ]
+        },
+        {
+          "name": "kafkaJaasInitializer",
+          "parameterTypes": []
+        },
+        {
+          "name": "kafkaAdmin",
+          "parameterTypes": [
+            "org.springframework.boot.kafka.autoconfigure.KafkaConnectionDetails"
+          ]
+        },
+        {
+          "name": "kafkaRetryTopicConfiguration",
+          "parameterTypes": [
+            "org.springframework.kafka.core.KafkaTemplate"
+          ]
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Admin",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Cleanup",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Consumer",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$IsolationLevel",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Jaas",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Listener",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Listener$Type",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Producer",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Properties",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Retry",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Retry$Topic",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Retry$Topic$Backoff",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Security",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Ssl",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Streams",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaProperties$Template",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true,
+      "allDeclaredFields": true
+    },
+    {
+      "type": "org.springframework.boot.autoconfigure.condition.OnClassCondition",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.autoconfigure.condition.OnBeanCondition",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.autoconfigure.condition.OnPropertyCondition",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.autoconfigure.condition.OnThreadingCondition",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.kafka.annotation.KafkaListenerConfigurationSelector",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        },
+        {
+          "name": "selectImports",
+          "parameterTypes": [
+            "org.springframework.core.type.AnnotationMetadata"
+          ]
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.kafka.annotation.KafkaBootstrapConfiguration",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        },
+        {
+          "name": "registerBeanDefinitions",
+          "parameterTypes": [
+            "org.springframework.core.type.AnnotationMetadata",
+            "org.springframework.beans.factory.support.BeanDefinitionRegistry"
+          ]
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.kafka.annotation.KafkaListenerAnnotationBeanPostProcessor",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.kafka.config.KafkaListenerEndpointRegistry",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.apache.kafka.common.serialization.StringDeserializer",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true
+    },
+    {
+      "type": "org.apache.kafka.common.serialization.StringSerializer",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true
+    },
+    {
+      "type": "org.springframework.kafka.listener.ContainerProperties$AckMode",
+      "allDeclaredFields": true,
+      "allPublicMethods": true
+    },
+    {
+      "type": "org.springframework.kafka.listener.ContainerProperties",
+      "allPublicMethods": true
+    },
+    {
+      "type": "sun.security.provider.ConfigFile",
+      "allDeclaredConstructors": true,
+      "allPublicMethods": true
+    },
+    {
+      "type": "org.springframework.context.annotation.ConfigurationClassParser$DefaultDeferredImportSelectorGroup",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration",
+      "allDeclaredConstructors": true,
+      "allDeclaredMethods": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaAnnotationDrivenConfiguration$EnableKafkaConfiguration",
+      "allDeclaredConstructors": true
+    },
+    {
+      "type": "org.springframework.boot.kafka.autoconfigure.KafkaStreamsAnnotationDrivenConfiguration",
+      "allDeclaredConstructors": true,
+      "allDeclaredMethods": true
+    },
+    {
+      "type": {
+        "proxy": [
+          "org.springframework.boot.autoconfigure.AutoConfiguration"
+        ]
+      }
+    },
+    {
+      "type": {
+        "proxy": [
+          "org.springframework.boot.autoconfigure.condition.ConditionalOnClass"
+        ]
+      }
+    },
+    {
+      "type": {
+        "proxy": [
+          "org.springframework.boot.context.properties.EnableConfigurationProperties"
+        ]
+      }
+    },
+    {
+      "type": "org.springframework.boot.context.properties.EnableConfigurationPropertiesRegistrar",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.context.properties.ConfigurationPropertiesBinder$ConfigurationPropertiesBinderFactory",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.context.properties.BoundConfigurationProperties",
+      "methods": [
+        {
+          "name": "<init>",
+          "parameterTypes": []
+        }
+      ]
+    },
+    {
+      "type": "org.springframework.boot.validation.beanvalidation.MethodValidationExcludeFilter",
+      "methods": [
+        {
+          "name": "byAnnotation",
+          "parameterTypes": [
+            "java.lang.Class"
+          ]
+        }
+      ]
+    },
+    {
+      "type": {
+        "proxy": [
+          "org.springframework.context.ConfigurableApplicationContext",
+          "org.springframework.boot.test.context.assertj.AssertableApplicationContext"
+        ]
+      }
+    }
+  ],
+  "resources": [
+    {
+      "glob": "org/springframework/boot/kafka/autoconfigure/*.class"
+    },
+    {
+      "glob": "org/springframework/kafka/annotation/*.class"
+    },
+    {
+      "glob": "org/springframework/context/annotation/*.class"
+    },
+    {
+      "glob": "META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports"
+    },
+    {
+      "glob": "META-INF/spring-autoconfigure-metadata.properties"
+    },
+    {
+      "glob": "org/springframework/boot/context/properties/EnableConfigurationPropertiesRegistrar.class"
+    }
+  ]
+}

--- a/metadata/org.springframework.boot/spring-boot-kafka/index.json
+++ b/metadata/org.springframework.boot/spring-boot-kafka/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "4.1.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-kafka/$version$/spring-boot-kafka-$version$-sources.jar",
+    "repository-url" : "https://github.com/spring-projects/spring-boot",
+    "test-code-url" : "https://github.com/spring-projects/spring-boot/tree/v$version$/module/spring-boot-kafka/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-kafka/$version$/spring-boot-kafka-$version$-javadoc.jar",
+    "description" : "Spring Boot Kafka provides auto-configuration and support classes for integrating Spring Boot applications with Apache Kafka through Spring for Apache Kafka. It configures Kafka producers, consumers, listener containers, transactions, metrics, Kafka Streams, and Testcontainers connection details based on Spring Boot properties and user-defined beans.",
+    "test-version" : "4.0.2",
+    "tested-versions" : [
+      "4.1.0"
+    ],
+    "allowed-packages" : [
+      "org.springframework.boot.kafka"
+    ]
+  },
+  {
     "metadata-version" : "4.0.2",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-kafka/$version$/spring-boot-kafka-$version$-sources.jar",
     "repository-url" : "https://github.com/spring-projects/spring-boot",

--- a/stats/org.springframework.boot/spring-boot-kafka/4.1.0/execution-metrics.json
+++ b/stats/org.springframework.boot/spring-boot-kafka/4.1.0/execution-metrics.json
@@ -1,0 +1,115 @@
+{
+  "fix_ni_run:2026-06-11": {
+    "timestamp": "2026-06-11T04:13:23.659751Z",
+    "library": "org.springframework.boot:spring-boot-kafka:4.1.0",
+    "starting_commit": "b37edfea3f9a1266193ca1b9ab92d961d3a11559",
+    "ending_commit": "4590f43269ca7b1ac11bc743fdce1d9f354f3bb3",
+    "previous_library": "org.springframework.boot:spring-boot-kafka:4.0.2",
+    "strategy_name": "library_update_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "4.1.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 2547,
+          "missed": 832,
+          "ratio": 0.753773,
+          "total": 3379
+        },
+        "line": {
+          "covered": 592,
+          "missed": 256,
+          "ratio": 0.698113,
+          "total": 848
+        },
+        "method": {
+          "covered": 226,
+          "missed": 135,
+          "ratio": 0.626039,
+          "total": 361
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "4.0.2",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 2488,
+          "missed": 824,
+          "ratio": 0.751208,
+          "total": 3312
+        },
+        "line": {
+          "covered": 580,
+          "missed": 252,
+          "ratio": 0.697115,
+          "total": 832
+        },
+        "method": {
+          "covered": 223,
+          "missed": 133,
+          "ratio": 0.626404,
+          "total": 356
+        }
+      }
+    },
+    "library_preparation_preflight": {
+      "status": "degraded",
+      "action": "no_action",
+      "issue_number": 8357,
+      "issue_label": "fails-native-image-run",
+      "library": "org.springframework.boot:spring-boot-kafka:4.1.0",
+      "summary": "Library preparation preflight did not produce usable advisory setup.",
+      "deterministic_setup": [],
+      "agent_guidance": "",
+      "risks": [],
+      "applied_setup": [],
+      "input_evidence": [
+        {
+          "name": "issue",
+          "summary": "#8357 [Automation] native-image run fails for org.springframework.boot:spring-boot-kafka:4.1.0; label=fails-native-image-run; body_chars=8015"
+        },
+        {
+          "name": "existing_tests",
+          "summary": "6 existing test file path(s) included"
+        }
+      ],
+      "current_library": "org.springframework.boot:spring-boot-kafka:4.0.2",
+      "new_version": "4.1.0",
+      "failure_reason": "FileNotFoundError: [Errno 2] No such file or directory",
+      "model": "oca/gpt-5.5",
+      "prompt_path": "library-preflight-prompt.txt",
+      "session_log_path": "/home/jovan/Work/graalvm-reachability-metadata/forge/logs/org.springframework.boot:spring-boot-kafka:4.1.0/library-preparation-preflight/pi-generation-2026-06-11T03-49-19-573569Z.log"
+    },
+    "metrics": {
+      "input_tokens_used": 0,
+      "cached_input_tokens_used": 0,
+      "output_tokens_used": 0,
+      "iterations": 0,
+      "cost_usd": 0.0,
+      "tested_library_loc": 592,
+      "metadata_entries": 141,
+      "previous_library_metadata_entries": 141,
+      "code_coverage_percent": 69.81,
+      "previous_library_coverage_percent": 69.71
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.springframework.boot/spring-boot-kafka/4.0.2/src/test/java/org_springframework_boot/spring_boot_kafka/Spring_boot_kafkaTest.java",
+      "metadata_file": "metadata/org.springframework.boot/spring-boot-kafka/4.1.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.springframework.boot/spring-boot-kafka/4.1.0/stats.json
+++ b/stats/org.springframework.boot/spring-boot-kafka/4.1.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "4.1.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 2547,
+        "missed" : 832,
+        "ratio" : 0.753773,
+        "total" : 3379
+      },
+      "line" : {
+        "covered" : 592,
+        "missed" : 256,
+        "ratio" : 0.698113,
+        "total" : 848
+      },
+      "method" : {
+        "covered" : 226,
+        "missed" : 135,
+        "ratio" : 0.626039,
+        "total" : 361
+      }
+    }
+  } ]
+}

